portal/server/companies-api.js

```diff
@@ -0,0 +1,56 @@
+const { isValidCompanyRegistrationNumber } = require('@ukef/dtfs2-common');
+
+const axios = require('axios');
+
+require('dotenv').config();
+
+const { PORTAL_API_URL } = process.env;
+
+const getCompanyByRegistrationNumber = async (registrationNumber, token) => {
+  try {
+    if (!registrationNumber) {
+      return {
+        errorMessage: 'Enter a Companies House registration number',
+      };
+    }
+
+    if (!isValidCompanyRegistrationNumber(registrationNumber)) {
+      return {
+        errorMessage: 'Enter a valid Companies House registration number',
+      };
+    }
+
+    const { data } = await axios.get(`${PORTAL_API_URL}/v1/companies/${registrationNumber}`, {
+      headers: {
+        Authorization: token,
+        'Content-Type': 'application/json',
+      },
+    });
+
+    return { company: data };
+  } catch (error) {
+    console.error(`Error calling Portal API 'GET /companies/:registrationNumber': %o`, error);
+    switch (error?.response?.status) {
+      case 400:
+        return {
+          errorMessage: 'Enter a valid Companies House registration number',
+        };
+      case 404:
+        return {
+          errorMessage: 'No company matching the Companies House registration number entered was found',
+        };
+      case 422:
+        return {
+          errorMessage: 'UKEF can only process applications from companies based in the UK',
```

Comment:
is this the only time it returns a 422? :)

Reply:
It should be, yes :)